daemon/cmd/kube_proxy_replacement_test.go

```diff
@@ -71,44 +128,7 @@ func (s *KubeProxySuite) TestDetectDevices(c *C) {
 	defer func() { c.Assert(testNetNS.Close(), IsNil) }()
 	defer func() { c.Assert(netns.Set(s.currentNetNS), IsNil) }()
 
-	// 1. No devices = impossible to detect
-	c.Assert(detectDevices(true, true), NotNil)
-
-	// 2. No devices, but no detection is required
-	c.Assert(detectDevices(false, false), IsNil)
-
-	// 3. Direct routing mode, should find dummy0 for both opts
-	c.Assert(createDummy("dummy0", "192.168.0.1/24"), IsNil)
-	c.Assert(createDummy("dummy1", "192.168.1.2/24"), IsNil)
-	c.Assert(createDummy("dummy2", "192.168.2.3/24"), IsNil)
-	option.Config.EnableIPv4 = true
-	option.Config.EnableIPv6 = false
-	option.Config.Tunnel = option.TunnelDisabled
-	node.SetK8sNodeIP(net.ParseIP("192.168.0.1"))
-	c.Assert(detectDevices(true, true), IsNil)
-	c.Assert(option.Config.Devices, checker.DeepEquals, []string{"dummy0"})
-	c.Assert(option.Config.DirectRoutingDevice, Equals, "dummy0")
-
-	// 4. dummy1 should be detected too
-	c.Assert(addDefaultRoute("dummy1", "192.168.1.1"), IsNil)
-	c.Assert(detectDevices(true, true), IsNil)
-	sort.Strings(option.Config.Devices)
-	c.Assert(option.Config.Devices, checker.DeepEquals, []string{"dummy0", "dummy1"})
-	c.Assert(option.Config.DirectRoutingDevice, Equals, "dummy0")
-
-	// 5. Enable IPv6, dummy1 should not be detected, as no default route for
-	// ipv6 is found
-	option.Config.EnableIPv6 = true
-	c.Assert(detectDevices(true, true), IsNil)
-	c.Assert(option.Config.Devices, checker.DeepEquals, []string{"dummy0"})
-	c.Assert(option.Config.DirectRoutingDevice, Equals, "dummy0")
-
-	// 6. Set random NodeIP, only dummy1 should be detected
-	option.Config.EnableIPv6 = false
-	node.SetK8sNodeIP(net.ParseIP("192.168.34.1"))
-	c.Assert(detectDevices(true, true), IsNil)
-	c.Assert(option.Config.Devices, checker.DeepEquals, []string{"dummy1"})
-	c.Assert(option.Config.DirectRoutingDevice, Equals, "dummy1")
+	act()
```

Comment:
Is it leftover?

Reply:
Which one? `act()`? This function abstracted the namespace setup and `act` is the actual test function invoked in the new fresh namespace. Perhaps it could use a better name?

Reply:
Where is it defined?

Reply:
The diff looks a bit funny as I moved a chunk of code around. Perhaps have a look at the whole file? I added `withFreshNetNS` function that takes a function as argument and it takes care of setting up and tearing down the namespace around the `act()` function call.

Reply:
Aha, I see it now. Maybe s/act/test/?